packages/clay-css/src/scss/functions/_global-functions.scss

```diff
@@ -53,25 +53,29 @@
 
 	$newMap: $map1;
 
-	@each $key, $value in $map2 {
-		@if (type-of($value) == map) {
-			$newMap: map-merge(
-				$newMap,
-				(
-					$key: map-deep-merge(map-get($newMap, $key), $value),
-				)
-			);
-		} @else {
-			$newMap: map-merge(
-				$newMap,
-				(
-					$key: $value,
-				)
-			);
+	@if (map-get($map2, merge) != false) {
```

Comment:
Looking at the issue it seems to me that the developer wants to prevent the class from being rendered in the final build, if that's the intention wouldn't that be the same thing as adding the `disabled` property?

Reply:
It's almost the same. A `disabled` (we have `enabled` atm) property would prevent a dev from reusing the class name. It would not output any CSS for `.btn-primary` for example. This `merge: false` tells Clay not to merge the values I set for `.btn-primary` with Clay's.

```scss
$btn-primary: (
    merge: false,
    background-color: orange,
);
```
would only output:
```css
.btn-primary {
    background-color: orange;
}
```
instead of:
```css
.btn-primary {
    background-color: orange;
    border-color: transparent;
    color: #fff;
}
.btn-primary:hover {
    background-color: #0053f0;
    border-color: transparent;
    color: #fff;
}
/* etc */
```

I think this is the better way, eventually we should get rid of the `enabled` attribute.

Reply:
I see, I don't understand if this is really something that would help in most use cases, it's just a little weird because it seems to be a very exclusive thing to remove all nested statements from the output while keeping the first level.

For me it still makes sense to make an `enabled`, because if the person doesn't want everything from Clay he can just disable some classes or variations of a component, for example he can import only what he wants but he can't ignore the classes of a file , I still see value in `enabled` as a way to "remove" specific classes from a file.

I think avoiding nested merge is just a way to go with it to keep the first level style. It seems to me that setting `null` seems more intuitive in subclasses a variable to say it doesn't exit on output.

Reply:
The `merge: false` also removes the class from the file. You just need to:

```scss
$btn-primary: (
   merge: false,
);
```
We would have no `.btn-primary {}` in the output.

Avoiding the merge is a QoL improvement for a dev that just wants to start styling `.btn-primary` without baggage from Clay. This isn't really needed, but thought it was a nice to have. I'm ok with adding it later if it gets requested.

Reply:
I see, yeah this looks like so the same function as `enabled`, I worry about this is that this would be the same thing as making the classes disabled and the developer adding `.btn-primary {...}` without any Clay resource.

I don't really like the idea of having this because the developer at the end is just overriding:

```sass
$btn-primary: (
   merge: false,
   background-color: orange,
);

// vs

$btn-primary: (
   enabled: false,
);

.btn-primary {
   background-color: orange,
}
```

This to me is two more different ways of offering this and not exactly a feature, I prefer the second option because it's explicit that you don't want a variance in the final output.

The first option wraps a lot of logic on top of the Clay only to end up with the same output as the second option, this easily bloats the Clay with resources it wouldn't need.

Reply:
Yeah it would essentially be a duplicate of the second option. The only difference is that in the first option we have access to Clay's Sass api. It will be useful when/if we implement design tokens, it will translate directly to CSS. The second option will require the dev to update their `.btn-primary` CSS to be able to take advantage of our tokens.

Reply:
Hm yeah that sounds like a good point for that, what would that look like directly?

Reply:
We would start with some JSON and convert it to Sass variables.

```json
{
  "btn-primary": {
    "merge": {
      "value": "false",
      "type": "other"
    },
    "background-color": {
      "value": "orange",
      "type": "color"
    },
    "border-color": {
      "value": "transparent",
      "type": "color"
    },
    "color": {
      "value": "{$white}",
      "type": "color"
    }
  }
}
```
would output:
```scss
$btn-primary: (
  merge: false,
  background-color: orange,
  border-color: transparent,
  color: $white,
);
```
The JSON format comes from https://www.figma.com/community/plugin/843461159747178978/Figma-Tokens. This would allow a designer using a Figma template we have put together to modify `.btn-primary` in Figma and output it for consumption by Clay. `merge` lets them start over if they wish or they can piggy back off our styles. `enabled` completely takes away the option of using `.btn-primary`.

Reply:
Yeah makes sense to me, I think that's a good point in favor of `merge` here. Now I'm not sure if `enabled` is equivalent to `merge`, so it looks like `merge` will be used for other purposes whereas `enabled` will just remove it from the final output. Feel free to go ahead with it.

Reply:
I'm going to merge it, thanks @matuzalemsteles ! I'm just going to note that we can also do:
```scss
$btn-primary: (
    focus: (
        merge: false,
    ),
);
```
and it won't output the focus styles for `.btn-primary`.